tests/test_readLineList.py

```diff
@@ -92,6 +98,48 @@ def testLamps(self):
         assert len(expect) == 2  # Two lines are Argon
         self.assertLineList(lineList, expect)
 
+    def testLampsHgCd(self):
+        """Test that we can select Mercury-Cadmium lines by lamp"""
+        metadata = PropertyList()
+        metadata.set("W_AITNEO", 0)  # Neon
+        metadata.set("W_AITARG", 0)  # Argon
+        metadata.set("W_AITKRY", 0)  # Krypton
+        metadata.set("W_AITHGC", 1)  # Mercury-Cadmium
+        self.config.restrictByLamps = True
+        lineList = self.makeLineList(metadata=metadata)
+        expect = [ll for ll in self.contents if ll.description in ["ArI", "HgI", "HgII"]]
+        assert len(expect) == 4  # Two lines are Mercury, two are Argon
+        self.assertLineList(lineList, expect)
+
+    def testfilterByLampElements(self):
+        """Tests that the correct lines are selected
+        for the given line elements,
+        even when the element names start with the same characters
+        """
+        task = ReadLineListTask()
+        lines = [
+            Line(123.456, 1000, "HI", 0),
+            Line(456.789, 1000, "HII", 0),
+            Line(567.890, 999, "ArI", 0),
+            Line(890.123, 123, 'HgII', 0),
+            Line(901.123, 14, 'HgI', 0),
+        ]
+
+        filteredLines = task.filterByLampElements(lines, ['H'])
+        self.assertLineList(filteredLines, lines[0:2])
```

Comment:
Please add a check on the results of `task.filterByLampElements(lines, ['HI'])`. There should be one result, but will be two.